core/sodasql/scan/scan_builder.py

```diff
@@ -19,13 +19,95 @@
 from sodasql.scan.scan import Scan
 from sodasql.scan.scan_yml import ScanYml
 from sodasql.scan.warehouse_yml import WarehouseYml
-from sodasql.scan.warehouse_yml_parser import read_warehouse_yml_file
+from sodasql.scan.warehouse_yml_parser import WarehouseYmlParser, read_warehouse_yml_file
 from sodasql.soda_server_client.soda_server_client import SodaServerClient
 from sodasql.scan.failed_rows_processor import FailedRowsProcessor
 
+
 logger = logging.getLogger(__name__)
 
 
+def build_warehouse_yml_parser(
```

Comment:
I tried to reuse as much as possible from the Scan class. Took out some pieces and put them in separate functions, so that they can be used in different locations more easily